### ISSUE TITLE
ReDim Preserve fails to preserve when the LBound is redimmed

### ISSUE THREAD

**Issue by fafalone** [twinbasic/twinbasic#1725]
**Describe the bug**
The values will be reset to zero. This is silent without raising an error, so I believe it should be counted as a bug, even though VB6 raises an error, since tB successfully changes the bounds, just not preserving the values.

**To Reproduce**
```
    Dim a%()

    ReDim a(2 To 10)
    a(9) = -1
    ReDim Preserve a(2 To 9)
    
    Debug.Print a(9)
  ```
  prints -1, but change it to `ReDim Preserve a(3 To 9)` and it will print 0, even though a(9) was never out of bounds.

**Expected behavior**
While I'd much prefer the `Preserve` functionality to work, it should at least raise an error like VB6 if that's impractical.

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Windows 10
 - twinBASIC compiler version 0.15.423

**Additional context**

**Comment by Megaseso** [comment]
Hello. I'm new to Twinbasic, and the truth is that I really like it.
I have been testing the Redim Preserve function, which also interested me, and I see that it works correctly as long as you do not modify the minimum value of the initial redim. That is, you can do redeem (5 to 20) and then redeem preserve (5 to 50) or (5 to 10), but if you modify the 5, you lose the stored values.

**Comment by dmrvb** [comment]
Can you check what happens in VB6??
I'm not sure but I think VB6 behaves in the same way and tB is currently trying to be 100% VB compatible, warts and all..
and yes, tB is looking rather good.

**Comment by GCuser99** [comment]
In VBA, you can only change the upper bound with Redim Preserve. Modifying the lower bound throws a "Subscript out of range".

**Comment by Megaseso** [comment]
That's right. The same thing happens in vb6. The lower value of the matrix should always be the same. The largest can be higher or lower without error.

**Comment by fafalone** [comment]
Whether to allow what would be an error in VB is pretty much case-by-case; I don't see this as a scenario where it's plausible compatibility with someone's app depends on raising the error here.

**Comment by wqweto** [comment]
There are many edge cases where VB6 raises error and TB chose to emulate the same (annoying) behavior so this one is another such case IMO.

**Comment by fafalone** [comment]
There's also places where tB allows something that would be an error. Like passing a Long to a method expecting a UDT not being a type mismatch in tB but rather taken as a pointer to that UDT. 

I don't think we want an absolute prohibition on allowing what would be invalid in VB6 or we'd lose great features like that.

**Comment by Megaseso** [comment]
For many years I have always used "redim preserve" to extend the upper limit but never to increase or decrease the lower limit.

**Comment by WaynePhillipsEA** [comment]
Fixed in BETA 699, thanks
